sw/host/opentitantool/src/command/rsa.rs

```diff
@@ -16,6 +19,17 @@ use opentitanlib::crypto::rsa::{
 use opentitanlib::crypto::sha256::Sha256Digest;
 use opentitanlib::util::parse_int::ParseInt;
 
+/// Load a public key, or a private key and extract a public key from it
+fn load_pub_or_priv_key(path: &PathBuf) -> Result<RsaPublicKey> {
+    match RsaPublicKey::from_pkcs1_der_file(path) {
+        Ok(key) => Ok(key),
+        Err(_) => match RsaPrivateKey::from_pkcs8_der_file(path) {
+            Ok(key) => Ok(RsaPublicKey::from_private_key(&key)),
+            Err(e) => Err(e),
+        },
+    }
```

Comment:
You can replace the inner `match` with `?`:

```rust
fn load_pub_or_priv_key(path: &PathBuf) -> Result<RsaPublicKey> {
    match RsaPublicKey::from_pkcs1_der_file(path) {
        Ok(key) => Ok(key),
        Err(_) => Ok(RsaPublicKey::from_private_key(RsaPrivateKey::from_pkcs8_der_file(path)?)),
    }
}
```

You can then de-duplicate the `Ok()` invocations:

```rust
fn load_pub_or_priv_key(path: &PathBuf) -> Result<RsaPublicKey> {
    Ok(match RsaPublicKey::from_pkcs1_der_file(path) {
        Ok(key) => key,
        Err(_) => RsaPublicKey::from_private_key(RsaPrivateKey::from_pkcs8_der_file(path)?),
    })
}
```

Alternatively, you can return early for a public key (I personally find this a bit nicer but it's subjective):

```rust
fn load_pub_or_priv_key(path: &PathBuf) -> Result<RsaPublicKey> {
    if let Ok(key) = RsaPublicKey::from_pkcs1_der_file(path) {
        return Ok(key);
    }
    Ok(RsaPublicKey::from_private_key(RsaPrivateKey::from_pkcs8_der_file(path)?))
}
```

Reply:
Thanks, I went for the early return, I also find it nicer.